Server/src/lobby.erl

```diff
@@ -0,0 +1,115 @@
+	
+	%This case happens when the message recieved contains a command for simulation the diagram
+	{command, Creator_Socket, {Did, Message_request}} -> 
+	  Pid = find_diagram(Did, Diagrams),
+	  %Sends the message to the diagram coordinator.
+	  Pid ! {Message_request, self()},
+	  %Confirmation that the diagram coordinator received the message
+	  receive
+	  	ok -> ok
+	  end,
+	  %Receives the result from the diagram coordinator and sends it to the client
+	  loop(Creator_Socket, Password, Members, Diagrams, Lobby_ID);
+	  
+	%This case happens when a previous message has been readded to the "queue" or if there were to previous messages to step back to.
+	{previous_confirmation, Did, Message} -> 
+	  %Turns the message we want to send into binary.
+	  Format_result = io_lib:format("~p", [{Did, previous_confirmation, Message}]),
+	  %Sends it to the client
+	  send_messages(Members, [Format_result ++ "~"]),
+	  loop(Creator_Socket, Password, Members, Diagrams, Lobby_ID);
+	
+	%This case happens when there is no more messages in the diagram and the user tries to simulate the next message
+	{simulation_done, Did, Message_number} -> 
+	  %Turns the result into binary
+	  Format_result = io_lib:format("~p", [{Did, simulation_finished, Message_number}]),
+      %Sends it to the client
+	  send_messages(Members, [Format_result ++ "~"]),
+	  loop(Creator_Socket, Password, Members, Diagrams, Lobby_ID);
+	  
+	%This case happens when a message has been sent in a diagram and the info is supposed to be sent to the client
+	{message_sent, Did, From, To, Message, Message_number} -> 
+	  %Turns the result into binary
+	  %The character ~ is used as the stop character for when the client should stop reading from the tcp connection
+	  Format_result = io_lib:format("~p", [{Did, From, To, Message, Message_number}]) ++ "~",
+      %Sends it to the client
+	  send_messages(Members, [Format_result]),
+	  loop(Creator_Socket, Password, Members, Diagrams, Lobby_ID);
+	  
+	%This case happens when there are messages to print to the execution log
+	{Did, print_information, Msg}->
+	  %Turns the message into binary
+	  Format_result = io_lib:format("~p", [{Did, print_information, Msg}]),
+      %Sends the result to client
+	  send_messages(Members, [Format_result ++ "~"]),
+	  loop(Creator_Socket, Password, Members, Diagrams, Lobby_ID)
+  end.
+  
+%Sends the classes of all given diagrams to the given user
+send_classes(_, []) -> done;
+send_classes(Socket, [{Diagram_ID, Class_names, _}|Diagrams]) -> 
+  Format_result = io_lib:format("~p", [{Diagram_ID, Class_names}]) ++ "~",
+  gen_tcp:send(Socket, Format_result),
+  send_classes(Socket, Diagrams).
+  
+%Finds the correct diagram from the given list
+find_diagram(_, []) -> not_created;
+find_diagram(Diagram_id,[{Diagram_id, _ ,Pid} | _]) -> Pid;
+find_diagram(Diagram_id,[_| Diagrams])  -> find_diagram(Diagram_id, Diagrams).  
+  
+%Finds the given member in the given list of members
```

Comment:
The name is correct but I think it's to explicit. Since the functionality is 'find'. _member is not required. More generic = better!

Reply:
but isn't it better to be more percise in the naming? since the function is finding a member of the lobby.  Just having 'find' could be confusing and present the question "find what?".